### PR TITLE
[FIX] account: always create an alias for sales/purchases journals

### PR DESCRIPTION
**Pull request by william-andre**
This will make the onboarding easier, and we won't propose to send a
mail to `False@mycompany.com`.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_journal.py [hunk 1]
```diff
@@ -523,7 +523,7 @@ def create(self, vals):
 
         journal = super(AccountJournal, self.with_context(mail_create_nolog=True)).create(vals)
 
-        if 'alias_name' in vals:
+        if journal.type in ('sale', 'purchase'):
```
Comment (smetl) [inline addons/account/models/account_journal.py:526]: Are you sure it's not breaking anything if `'alias_name'` is not in vals ?

Comment (william-andre) [reply addons/account/models/account_journal.py:526]: https://github.com/odoo/odoo/blob/5e684773fca39bc15f99d7c83889818b351b424a/addons/account/models/account_journal.py#L371-L373

then

https://github.com/odoo/odoo/blob/5e684773fca39bc15f99d7c83889818b351b424a/addons/account/models/account_journal.py#L330-L334

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by smetl** [comment]
@robodoo r+

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]
Seems like you should probably better handle alias update in your custom ``_update_mail_alias`` method. Not sure what you are trying to do but this reads more like a hack than a fix.

Thanks !

**Comment by tde-banana-odoo** [comment]
@robodoo r-

**Comment by william-andre** [comment]
> Seems like you should probably better handle alias update in your custom `_update_mail_alias` method. Not sure what you are trying to do but this reads more like a hack than a fix.
> 
> Thanks !

@tde-banana-odoo 
TL;DR: I'm just trying to create an alias when it is possible without forbidding the user to create a journal because a mail alias could not be created.

______

The CI encountered two issues: one alias was declared in `hr_expense`, and a `account.journal` with the same name as the alias was impossible.
The other issue was a test creating a few journals all with the same name, and the alias created was the same for all journals.

While I agree this looks like a hack, it is because specialized errors are seen like evil by many people at Odoo.
I would have gladly done
```python
class DuplicateAliasError(UserError):
    pass
```

```python
        if sanitized_name in [catchall_alias, bounce_alias] or self.search_count(domain):
            DuplicateAliasError(_('The e-mail alias is already used. Please enter another one.'))
```

```python
try:
    journal._update_mail_alias(vals)
except DuplicateAliasError as error:
    pass
```

But this doesn't work because it would require this commit in order to work https://github.com/odoo-dev/odoo/commit/52ef6517292f24de5468a61da730dbe20c193603 properly.

**Comment by tde-banana-odoo** [comment]
I think you should pre-process values locally. Having several times the same alias is forbidden and you should always ensure values given to create are correct. It is caller job to ensure what it gives it right, not try to hack a global behavior for some demo data.

You may define an uniquifier in your account module generating aliases + a suffix if you want to automatically generate aliases that are valid and not duplicates. Or set it to False if already taken as a pro process check. Note that void values are not considered in this unique constraint, so if you don't know what to do you can set them to False.

About error management: well I think this is not necessary as it is adding complexity while issue is that values given to alias mixin are not considered as correct.

Thanks !

**Comment by william-andre** [comment]
Will be fixed by https://github.com/odoo/odoo/pull/75370
